CorsixTH/Lua/dialogs/resizables/menu_list_dialogs/custom_campaign.lua

```diff
@@ -33,40 +33,41 @@ local col_scrollbar = {
 local details_width = 280
 
 --! Collect the campaign levels at the provided path
---!param path (str) File system path to search.
---!return (array) The found levels, with some basic information about each level.
-local function createCampaignList(path)
+--!param paths_table (array) File system paths to search.
+--!return (array) The found levels, with some basic information about each level, in alphabetical order.
+local function createCampaignList(paths_table)
   local campaigns = {}
-
-  for file in lfs.dir(path) do
-    if file:match("%.campaign$") then
-      local campaign_info, err = TheApp:readCampaignFile(file)
-      if not campaign_info then
-        print(err)
-      else
-        if campaign_info.levels and #campaign_info.levels > 0 then
-          campaigns[#campaigns + 1] = {
-            name = campaign_info.name,
-            tooltip = _S.tooltip.custom_campaign_window.choose_campaign,
-            no_levels = #campaign_info.levels,
-            path = file,
-            description = TheApp.strings:getLocalisedText(campaign_info.description,
-               campaign_info.description_table)
-          }
+  for _, folder in pairs(paths_table) do
+    for file in lfs.dir(folder) do
+      if file:match("%.campaign$") then
+        local campaign_info, err = TheApp:readCampaignFile(file)
+        if not campaign_info then
+          print(err)
         else
-          print("Warning: Loaded campaign that had no levels specified")
+          if campaign_info.levels and #campaign_info.levels > 0 then
+            campaigns[#campaigns + 1] = {
+              name = campaign_info.name,
+              tooltip = _S.tooltip.custom_campaign_window.choose_campaign,
+              no_levels = #campaign_info.levels,
+              path = file,
+              description = TheApp.strings:getLocalisedText(campaign_info.description,
+                 campaign_info.description_table)
+            }
+          else
+            print("Warning: Loaded campaign that had no levels specified")
+          end
         end
       end
     end
   end
+  table.sort(campaigns, function(a,b) return a.name < b.name end)
```

Comment:
We get duplicate names if the same campaign exists in both the user and the game directories.

Reply:
As this is the name from reading the file, it's already possible from two files in the same folder. Should they be given suffixes, like (2)?

Reply:
Numbered suffixes don't help tell them apart, adding the filename to the description of clashes doesn't look good.

Reply:
For now, ignore all campaigns with the same name except the first one, dump a log entry with the details, and give a window that says some campaigns were not added due to having duplicated names?